keras_nlp/models/llama/llama_layernorm.py

```diff
@@ -14,24 +14,36 @@
 from keras_nlp.backend import keras
 from keras_nlp.backend import ops
 
-# TODO: Should be replaced with LayerNormalization with `rms_scaling` param
-# https://github.com/keras-team/keras-core/pull/726
-
 
+# TODO: Deprecate this in favor of
+# `keras.layers.LayerNormalization(rms_scaling=True)` once Keras 2 support is
+# removed.
 class LlamaLayerNorm(keras.layers.Layer):
+    """A normalization layer for Llama that implements RMS normalization."""
+
     def __init__(self, epsilon=1e-6, **kwargs):
         super().__init__(**kwargs)
-        self.epsilon = epsilon
+        self._epsilon = epsilon
 
     def build(self, input_shape):
-        self.weight = self.add_weight(
+        self._dim = input_shape[-1]
+        self._weight = self.add_weight(
```

Comment:
Keep this public too. Update mistral instead.

Do our checkpoints still load fine if we call this scale? And name it scale? That's what Gemma does and it's a a better name.

Reply:
> Do our checkpoints still load fine if we call this scale? And name it scale?

I think Keras should work when the variable name in Python changes. AFAIK, Keras loads weights using the `name` field of the variable. So, changing that would break loading.

Reply:
I believe it is actually order based. Did you try it out? https://github.com/keras-team/keras/blob/97b082dfee2552fcad1a7c7ea0fac9c72943360c/keras/layers/layer.py#L1187-L1188

It's not the end of the world. Just for things like the optimize call [here](https://ai.google.dev/gemma/docs/distributed_tuning#finetune_with_imdb) we actually reference the scale name.

Reply:
Oh cool. Thanks for the references. Will try to change and check if I can still load the preset!

Reply:
I verified that weights loaded with the name change, so changes pushed. Let's see if the CI is also happy with it. Thanks!